### ISSUE TITLE
Pre-Signed URLs for non-GET Actions

### ISSUE THREAD

**Issue by SattaiLanfear** [Rhinofly/play-s3#43]
Originally posted in #20.

I need the ability to generate pre-signed urls for http-actions other than GET.  I'm writing a restful play server but I don't want the clients to send the play server the files to upload to S3 - I want the clients to be able to upload them directly to S3 using PUT requests, without giving the clients real credentials for AWS.

Ideally I'd have something along the lines of an endpoint in the play server that just says the equivalent of:

You want to upload hello-world.jpg?
PUT it here: https://...bucket-url.../hello-world.jpg?aws-required-auth-info-etc

Basically offloading the effort of doing the upload back to the client (a mobile app which has to do the upload at least once anyway).

Unfortunately it seems all the url signing methods are hard-coded to only do "GET".

**Comment by EECOLOR** [comment]
Checkout https://github.com/Rhinofly/play-s3/tree/6.0.0, it's available as `6.0.0-RC1`. Let me know if this solves your problem.

**Comment by SattaiLanfear** [comment]
I think it does actually.  I haven't had the chance to download it but browsing the code, it looks like I could access `S3.client.signer.signUrl` to create any url required now using the same configuration as the rest of the application.

Thanks.

**Comment by EECOLOR** [comment]
Ahh, yes. So I should add a method on the bucket to make that more easy I guess.
